swarm_conn.go

```diff
@@ -54,6 +54,11 @@ func (c *Conn) ID() string {
 // open notifications must finish before we can fire off the close
 // notifications).
 func (c *Conn) Close() error {
+	// for mock tests
+	if c == nil || c.conn == nil {
+		return nil
+	}
+
```

Comment:
???? This is really scary. Why?

Reply:
unnecessary after fixing a bug, will remove!